src/screens/ChoosePlanScreen/components/SubscriptionList.tsx

```diff
@@ -0,0 +1,249 @@
+import { Box, ScrollView, useColorMode } from 'native-base';
+import React, { useState } from 'react';
+import { View, TouchableOpacity, StyleSheet } from 'react-native';
+import CircleIconWrapper from 'src/components/CircleIconWrapper';
+import { wp } from 'src/constants/responsive';
+import { useSettingKeeper } from 'src/hooks/useSettingKeeper';
+import ArrowRightGreen from 'src/assets/images/right-green-icon.svg';
+import ArrowDownGreen from 'src/assets/images/down-green-icon.svg';
+import ArrowDownWhite from 'src/assets/images/down-white-icon.svg';
+import ArrowRightWhite from 'src/assets/images/right-white-icon.svg';
+import Text from 'src/components/KeeperText';
+import PlanDetailsCards from './PlanDetailsCards';
+import { KeeperApp } from 'src/models/interfaces/KeeperApp';
+import { useQuery } from '@realm/react';
+import { RealmSchema } from 'src/storage/realm/enum';
+import Buttons from 'src/components/Buttons';
+import Colors from 'src/theme/Colors';
+
+const SubscriptionList: React.FC<{
+  plans: any[];
+  currentPosition: number;
+  onChange?: any;
+  primaryCallback?: any;
+  isMonthly: boolean;
+  getButtonText?: any;
+}> = ({ plans, currentPosition, onChange, primaryCallback, isMonthly, getButtonText }) => {
+  const { colorMode } = useColorMode();
+  const isDarkMode = colorMode === 'dark';
+  const { planData } = useSettingKeeper();
+  const { subscription }: KeeperApp = useQuery(RealmSchema.KeeperApp)[0];
+  const [expandedIndex, setExpandedIndex] = useState<number | null>(null);
+
+  const [currentPositions, setCurrentPositions] = useState(
+    currentPosition !== 0 ? currentPosition : subscription.level - 1
+  );
+
+  const _onSnapToItem = (index) => {
+    setCurrentPositions(index);
+    onChange(index);
+  };
+
+  const toggleExpand = (index: number) => {
+    setExpandedIndex(expandedIndex === index ? null : index);
+  };
+
+  return (
+    <ScrollView showsVerticalScrollIndicator={false} contentContainerStyle={styles.container}>
+      {planData.map((plan, index) => {
+        const isExpanded = expandedIndex === index;
+        const isActive = index === currentPositions;
+
+        const matchedPlan = plans.find((p) => p.name.toLowerCase() === plan.title.toLowerCase());
+        const isPleb = plan.title.toLowerCase() === 'pleb';
+
+        const planDetails = isPleb
+          ? 'Free'
+          : isMonthly
+          ? matchedPlan?.monthlyPlanDetails?.price
+            ? 'US' + matchedPlan.monthlyPlanDetails.price + '/'
```

Comment:
The US here is wrong, it shows up even if the price shown is in any other currency